### PR TITLE
nic partitioning for SRIOV devices

### PR DESCRIPTION
**Pull request by lamek**

### CODE REVIEW

File: modules/nw-sriov-nic-partitioning.adoc [hunk 1]
```diff
@@ -0,0 +1,68 @@
+// Module included in the following assemblies:
+//
+// * networking/multiple_networks/configuring-sr-iov.adoc
+
+[id="nw-sriov-nic-partitioning_{context}"]
+= NIC partitioning for SR-IOV devices
+
+In some cases, you may want to split virtual functions (VFs) from the same physical function (PF) into multiple resource pools. For example, you may want some of the VFs with the default driver loaded, while the remaining VFs load with the `vfio-pci` driver. In such a deployment, the `pfNames` selector in your SriovNetworkNodePolicy Custom Resource can be used to specify a range of VFs for a pool using the following format: `<PFName>#<FirstVF>-<LastVF>`
```
Comment (jboxman) [inline modules/nw-sriov-nic-partitioning.adoc]: s/may/might/

File: modules/nw-sriov-nic-partitioning.adoc [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+For example, the following YAML shows the selector for an interface named netpf0 with VF 2 through 7:
```
Comment (jboxman) [inline modules/nw-sriov-nic-partitioning.adoc]: `netpf0`

File: modules/nw-sriov-nic-partitioning.adoc [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+[source,yaml]
+----
+pfNames: ["netpf0#2-7"]
+----
+
+* `netpf0` is the PF interface name.
+* `2` is the first VF index (0-based) that is included in the range.
+* `7` is the last VF index (0-based) that is included in the range.
+
+You can select VFs from the same PF by different policy CRs as long as the following requirements are met:
```
Comment (jboxman) [inline modules/nw-sriov-nic-partitioning.adoc]: s/by/by using/

Comment (jboxman) [inline modules/nw-sriov-nic-partitioning.adoc]: s/as long as/if/

File: modules/nw-sriov-nic-partitioning.adoc [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+* The `numVfs` should be identical for policies that select the same PF.
+* The VF index should be in in range of `0` to `<numVfs>-1`. For example, if you have a policy with `numVfs` 8, then the `<FirstVF>` should not be smaller than 0, and the `<LastVF>` should not larger than 7.
```
Comment (jboxman) [inline modules/nw-sriov-nic-partitioning.adoc]: s/in in/in/

Comment (jboxman) [inline modules/nw-sriov-nic-partitioning.adoc]: s/with `numVfs` 8/with `numVfs` set to `8`/

File: modules/nw-sriov-nic-partitioning.adoc [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+pfNames: ["netpf0#2-7"]
+----
+
+* `netpf0` is the PF interface name.
+* `2` is the first VF index (0-based) that is included in the range.
+* `7` is the last VF index (0-based) that is included in the range.
+
+You can select VFs from the same PF by different policy CRs as long as the following requirements are met:
+
+* The `numVfs` should be identical for policies that select the same PF.
```
Comment (jboxman) [inline modules/nw-sriov-nic-partitioning.adoc]: All s/should/must/ I think

File: modules/nw-sriov-nic-partitioning.adoc [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+* The VF index should be in in range of `0` to `<numVfs>-1`. For example, if you have a policy with `numVfs` 8, then the `<FirstVF>` should not be smaller than 0, and the `<LastVF>` should not larger than 7.
+* The VFs ranges in different policies should not overlap.
+* The `<FirstVF>` should not larger be than the `<LastVF>`.
+
+The following example illustrates NIC partitioning for an SR-IOV device.
+
+The policy `policy-net-1` defined a resource pool `net-1` which contains the VF 0 of PF netpf0, with the default VF driver. The policy `policy-net-1-dpdk` defined another resource pool `net-1-dpdk` which contains the VF 8 to 15 of PF netpf0, with the vfio VF driver.
```
Comment (jboxman) [inline modules/nw-sriov-nic-partitioning.adoc]: s/which/that/

`netpf0`

`0`

`8`

`15`

`vfio` probably

**Review by jboxman — CHANGES_REQUESTED** [review]
@lamek, suggested some changes. In the future I want to go back and scrub all of SR-IOV, after the topics are split out.

**Comment by lamek** [comment]
@pliurh can you PTAL?

**Comment by pliurh** [comment]
LGTM

**Pull request merged**